### PR TITLE
Set the service status to FAILED if the MySQL install failed.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Added some comments to the hard to follow MySqlAppStatus tests.

### CODE REVIEW

File: src/nova/guest/mysql/MySqlApp.cc [hunk 1]
```diff
@@ -173,38 +174,35 @@ void MySqlApp::install_and_secure(AptGuest & apt, int memory_mb) {
...
     status->begin_mysql_install();
 
-    // You may wonder why there's no try / catch around this section which
-    // would change status to a failure state if something went wrong.
-    // The reason why is that the Reddwarf compute manager will time out the
-    // instance. If we added a catch here, it would be harder to test this
-    // ability of the Reddwarf compute manager. We may find a way to add error
-    // handling here anyway as it would speed up acknowledgement of failures,
-    // but the bigger priority is making sure the manager retains the ability
-    // to determine when a guest is misbehaving.
```
Comment (TimSimpsonR) [inline src/nova/guest/mysql/MySqlApp.cc:183]: I don't think I've ever encountered a more helpful comment. All mystery and intrigue of why I hadn't already done this dissipated in an instant!

**Comment by rnirmal** [comment]
+1
